src/github.com/matrix-org/go-neb/services/slackapi/slackapi.go

```diff
@@ -53,6 +56,20 @@ func (s *Service) OnReceiveWebhook(w http.ResponseWriter, req *http.Request, cli
 	w.WriteHeader(200)
 }
 
+// Register joins all configured rooms
+func (s *Service) Register(oldService types.Service, client *matrix.Client) error {
+	for _, mapping := range s.Hooks {
+		if _, err := client.JoinRoom(mapping.RoomID, "", ""); err != nil {
+			log.WithFields(log.Fields{
+				log.ErrorKey: err,
+				"room_id":    mapping.RoomID,
+				"user_id":    client.UserID,
+			}).Error("Failed to join room")
+		}
+	}
+	return nil
```

Comment:
As I understand it, for webhooks in Go-NEB, the last path component is the base64 encoded service ID. The path must begin with `/services/hooks` and I have tested that `/services/hooks/$HOOK_ID/$SERVICE_ID` is handled correctly. 

So you _can_ have multiple webhooks directed at a single service, and the `webhookEndpointURL` provided _is_ an example of a hook that will hit the service, but it does not include a `HookID`. 

So there are two options: 1. remove the hook ID as a concept, and just have one webhook per slackapi service or 2. just send the existing `webhookEndpointURL` without the hook ID and expect the client to fill it in.